docs/en/week07/07-3.md

```diff
@@ -0,0 +1,133 @@
+
+$$
+F(\boldsymbol{y}) = C(\boldsymbol{y},\boldsymbol{\tilde{y}}) + R(\boldsymbol{h})
+$$
+
+### Reconstruction Costs
+Below are the two examples of reconstruction energies:
+##### Real Valued Input:
+$$
+C(\boldsymbol{y},\boldsymbol{\tilde{y}}) = \Vert \boldsymbol{y}-\boldsymbol{\tilde{y}} \Vert^2 =  \Vert \boldsymbol{y}-Dec[Enc(\boldsymbol{y})] \Vert^2
+$$
+This is the square ecludian distance between $\boldsymbol{y}$ and $\boldsymbol{\tilde{y}}$.
+
+##### Binary Input:
+In the case of binary input, we can simply use binary cross-entropy
+
+$$
+C(\boldsymbol{y},\boldsymbol{\tilde{y}}) = - \sum_{i=1}^n{[\boldsymbol{y_i}\log(\boldsymbol{\tilde{y_i}}) + (1-\boldsymbol{y_i})\log(1-\boldsymbol{\tilde{y_i}})]}
+$$
+
+### Loss Function
+Average across all training samples of per sample loss function
+
+$$
+L(F(\cdot),\boldsymbol{Y}) = \frac{1}{m}\sum_{j=1}^m{\ell(F(\cdot),\boldsymbol{y}^{(j)})} \in \mathbb{R}
+$$
+We take the energy loss and try to push the energy down on $\boldsymbol{\tilde{y}}$
+$$
+\ell_{energy}(F(\cdot),\boldsymbol{y}) = F(\boldsymbol{y})
+$$
+
+
+### Use-cases
+The size of the hidden representation $\boldsymbol{h}$ obtained using these networks can be both smaller and larger than the input size. 
+
+If we choose a smaller $\boldsymbol{h}$, the network can be used for non-linear dimensionality reduction.
+
+In some situations it can be useful to have a larger than input $\boldsymbol{h}$, however, in this scenario, a plain autoencoder would collapse. In other words, since we are trying to reconstruct the input, the model is prone to copying all the input features into the hidden layer and passing it as the output thus essentially behaving as an identity function. This needs to be avoided as this would imply that our model fails to learn anything.
+
+To prevent the model from collapsing, we have to employ techniques that constrain the amount of region which can take zero or low energy values. These techniques can be some sort of regularization such as sparsity constraints, adding additional noise, or sampling.
+
+### Types of Auto-Encoders
+
+#### Denoising autoencoder - 
+
+We add some augmentation/corruption like Gaussian noise to an input sampled from the training manifold $\boldsymbol{\hat{y}}$ before feeding it into the model and expect the reconstructed input $\boldsymbol{\tilde{y}}$ to be similar to the original input $\boldsymbol{y}$.
+<center>
+<img src="{{site.baseurl}}/images/week07/07-3/DenoisingAutoEncoder.png" style="background-color:#DCDCDC;" /><br>
+<b>Figure 4:</b> Denoising Autoencoder Network architecture.
+</center>
+<!-- ![](https://i.imgur.com/WVcDLns.png) -->
+An important note: The noise added to the original input should be similar to what we expect in reality, so the model can easily recover from it.
+
+<center>
+<img src="{{site.baseurl}}/images/week07/07-3/DAEOutput.jpeg" style="background-color:#DCDCDC;" /><br>
+<b>Figure 5:</b> Network output.
```

Comment:
Use newer images from [this year's slides](https://drive.google.com/drive/u/1/folders/1cy7BHpZJQhkhbqwLShfq8UyYma11gGQa).

Reply:
Also, that's not quite the network's output. Please, describe the figure.

Reply:
Done